src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -443,4 +427,49 @@ public String computeResultDigest(ComputedFile computedFile) {
         }
         return resultDigest;
     }
+
+    // region Purge
+
+    /**
+     * Purge results from given task, especially its result folder.
+     *
+     * @param chainTaskId ID of the task to purge.
+     * @return {@literal true} if task folder has been deleted
+     * and task data has been cleared from this service;
+     * {@literal false} otherwise.
+     */
+    @Override
+    public boolean purgeTask(String chainTaskId) {
+        final String taskBaseDir = workerConfigService.getTaskBaseDir(chainTaskId);
+
+        resultInfoMap.remove(chainTaskId);
+        FileHelper.deleteFolder(taskBaseDir);
+
+        final boolean deletedInMap = !resultInfoMap.containsKey(chainTaskId);
+        final boolean deletedTaskFolder = !new File(taskBaseDir).exists();
+
+        boolean deleted = deletedInMap && deletedTaskFolder;
+        if (deletedTaskFolder) {
+            log.info("The result of the chainTaskId has been deleted [chainTaskId:{}]", chainTaskId);
+        } else {
+            log.warn("The result of the chainTaskId couldn't be deleted [chainTaskId:{}, deletedInMap:{}, " +
+                            "deletedTaskFolder:{}]",
+                    chainTaskId, deletedInMap, deletedTaskFolder);
+        }
+
+        return deleted;
+    }
+
+    /**
+     * Purge results from all known tasks, especially their result folders.
+     */
+    @Override
+    public void purgeAllTasksData() {
+        // `resultInfoMap.keyset()` needs to be wrapped.
+        // Otherwise, it'll be cleared when `resultInfoMap.clear` is called.
+        final Set<String> chainTaskIds = new HashSet<>(resultInfoMap.keySet());
+        resultInfoMap.clear();
+        chainTaskIds.forEach(chainTaskId -> FileHelper.deleteFolder(workerConfigService.getTaskBaseDir(chainTaskId)));
```

Comment:
At the end, we don't really know what went well in this code.

The following implementation would be a little bit slower but would reuse code and produce logging events on what went wrong:
```suggestion
        final Set<String> chainTaskIds = new HashSet<>(resultInfoMap.keySet());
        chainTaskIds.forEach(this::purgeTask);
```

Reply:
Good idea! Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/476/commits/eba9281658767b34761a605b905c4880fde82917.